pkg/node/node.go

```diff
@@ -309,5 +309,20 @@ func (n *Node) Marshal() ([]byte, error) {
 
 // Unmarshal parses the JSON byte slice and updates the node receiver
 func (n *Node) Unmarshal(data []byte) error {
```

Comment:
Is it a good idea to even allow Unmarshal to update an existing value? After the operation any private values may be out of sync with the updated public values.